src/parse.rs

```diff
@@ -561,31 +590,22 @@ mod tests {
     fn parse_vanished_test() {
         // VANISHED can appear if the user has enabled QRESYNC (RFC 7162), in response to
         // SELECT/EXAMINE (QRESYNC); UID FETCH (VANISHED); or EXPUNGE commands. In the first
-        // two cases the VANISHED respone will be a different type than the expected response
-        // and so goes into the unsolicited respones channel. In the last case, VANISHED is
-        // explicitly a response to an EXPUNGE command, but the semantics of EXPUNGE (one ID
-        // per response, multiple responses) vs VANISHED (a sequence-set of UIDs in a single
-        // response) are different enough that is isn't obvious what parse_expunge() should do.
-        // If we do nothing special, then the VANISHED response ends up in the unsolicited
-        // responses channel, which is at least consistent with the other cases where VANISHED
-        // can show up.
+        // two cases the VANISHED response will be a different type than expected
+        // and so goes into the unsolicited responses channel.
         let lines = b"* VANISHED 3\r\n";
         let (mut send, recv) = mpsc::channel();
         let resp = parse_expunge(lines.to_vec(), &mut send).unwrap();
-        assert!(resp.is_empty());
 
-        match recv.try_recv().unwrap() {
-            UnsolicitedResponse::Vanished { earlier, uids } => {
-                assert!(!earlier);
-                assert_eq!(uids.len(), 1);
-                assert_eq!(*uids[0].start(), 3);
-                assert_eq!(*uids[0].end(), 3);
-            }
-            what => panic!("Unexpected response in unsolicited responses: {:?}", what),
-        }
+        // Should have one UID response
+        assert!(!resp.is_empty());
+        let mut uids = resp.uids().unwrap();
+        assert_eq!(Some(3), uids.next());
+        assert_eq!(None, uids.next());
```

Comment:
Should this also assert that `seqs()` is empty?

Reply:
Ah, yes!